utils/filecache/fsfilecache.go

```diff
@@ -0,0 +1,78 @@
+package filecache
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ARM-software/golang-utils/utils/commonerrors"
+	"github.com/ARM-software/golang-utils/utils/filesystem"
+)
+
+// FsEntryProvider implements EntrySource by copying files or directories from a source filesystem into a cache filesystem.
+type FsEntryProvider struct {
+	fs            filesystem.FS
+	cachefs       filesystem.FS
+	basePath      string
+	cacheBasePath string
+}
+
+// FetchEntry locates the resource identified by `key` within the provider’s source filesystem under `basePath`,
+// copies it into `cacheDir` using `cacheFilesystem`, and returns the absolute path of the cached entry.
+func (p *FsEntryProvider) FetchEntry(ctx context.Context, key string) (string, error) {
+	srcPath := filesystem.FilePathJoin(p.fs, p.basePath, key)
+	if exists := p.fs.Exists(srcPath); !exists {
```

Comment:
just simplify this, it isn't a complex condition

```suggestion
	if !p.fs.Exists(srcPath) {
```